### PR TITLE
[FlexFlash] dynamic shapes

### PR DESCRIPTION
**Pull request by drisspg**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.13.0) (oldest at bottom):
* __->__ #171465

# Summary

Add a bunch of tests around dynamic shapes and create better error message for captured symbols in score-mod / mask mods since we need to do somehting similiar to aux_tensors to support this 

image

Since b200 ci is down

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

File: test/inductor/test_flex_flash.py [hunk 1]
```diff
@@ -995,6 +995,219 @@ def fn(x, weight):
...
+
+        self._run_dynamic_test(
+            seq_lens=[128, 256, 512],
+            block_mask_factory=block_mask_factory,
+            requires_grad=True,
+        )
+
+    def test_dynamic_gqa(self):
+        """Test GQA with dynamic sequence lengths."""
+        torch._dynamo.reset()
```
Comment (Skylion007) [inline test/inductor/test_flex_flash.py]: Which tests need dynamo resets and which don't?

Comment (drisspg) [reply test/inductor/test_flex_flash.py]: all of them are implicitly using dyanmo reset that invoke _run_dynamic: https://github.com/pytorch/pytorch/pull/171465/changes#diff-230988adbc86089cae8166b6341aff5b177b72a3728e81a3f1daad3ca1d925f8R1011

**Review by Skylion007 — APPROVED** [review]
